ksqldb-execution/src/main/java/io/confluent/ksql/execution/util/ComparisonUtil.java

```diff
@@ -76,9 +79,32 @@ private static boolean handleString(final Type operator, final SqlType right) {
     return right.baseType() == SqlBaseType.STRING;
   }
 
+  @SuppressWarnings({"checkstyle:BooleanExpressionComplexity", "checkstyle:LineLength"})
   private static boolean handleBoolean(final Type operator, final SqlType right) {
     return right.baseType() == SqlBaseType.BOOLEAN
-        && (operator == Type.EQUAL || operator == Type.NOT_EQUAL);
+        && (operator == Type.EQUAL || operator == Type.NOT_EQUAL
+            || operator == Type.IS_DISTINCT_FROM || operator == Type.IS_NOT_DISTINCT_FROM);
```

Comment:
nit: can we put all of these into a method so that we can repeat it below? (e.g. `isEqualityOperator(operator)`) if we do that we might even be able to dismiss the checkstyle suppressions

Reply:
Refactored